### ISSUE TITLE
SC1071 should not get checked if `-s` is specified

### ISSUE THREAD

**Issue by blueyed** [koalaman/shellcheck#974]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1071
- My shellcheck version (`shellcheck --version` or "online"): 0.4.6

`shellcheck -s sh` should use the "sh" dialect, and not look at the shebang line, possibly causing `SC1071`:

```sh
% shellcheck -s sh =(echo '#!/bin/zsh')

In /tmp/zshLclZZ0 line 1:
#!/bin/zsh
^-- SC1071: ShellCheck only supports sh/bash/dash/ksh scripts. Sorry!
```

**Comment by blueyed** [comment]
Just came across this again.

Any feedback on this?
Might make it a bit faster even when the shebang parsing would be skipped.

My use case is running it for zsh scripts occasionally.

I could like with a warning, but you could assume that if `--shell` is specified, this one should be just used.

**Comment by blueyed** [comment]
Related docs: https://github.com/koalaman/shellcheck/blame/10edba3ab82d58fb262a6ab3cc0245c0e6372259/shellcheck.1.md#L59-L63

The workaround would be for me to remove the shebang before calling shellcheck on it, but I would really like to avoid that.

I think the following needs to be skipped in case the shell option is given, but I do not know how:

https://github.com/koalaman/shellcheck/blob/10edba3ab82d58fb262a6ab3cc0245c0e6372259/src/ShellCheck/Parser.hs#L2964-L2966

**Comment by blueyed** [comment]
Awesome, thanks!

btw: is there any chance to get back support for Zsh?
What has been the reason to drop it in the first place?
I've only found ed56a83 with no explanations.

**Comment by koalaman** [comment]
You can follow the progress (or lack thereof) on  #809.

**Comment by blueyed** [comment]
Thanks!
